### PR TITLE
Add mkdir --mode

### PR DESCRIPTION
**Pull request by xavier**
It currently only support permissions given in octal format as I did not want to duplicate a significant amount of argument parsing code from chmod.  There's certainly an opportunity to create a `FilePermissionParser` class to be shared by these utilities but I'm not sure where it should belong.

In the process, I have updated mkdir to use LibMain.

### CODE REVIEW

File: Userland/Utilities/mkdir.cpp [hunk 1]
```diff
@@ -8,27 +8,35 @@
...
 #include <unistd.h>
 
-int main(int argc, char** argv)
+ErrorOr<int> serenity_main(Main::Arguments arguments)
 {
-    if (pledge("stdio cpath rpath", nullptr) < 0) {
-        perror("pledge");
-        return 1;
-    }
+    TRY(Core::System::pledge("stdio cpath rpath", nullptr));
```
Comment (bgianfo) [inline Userland/Utilities/mkdir.cpp:19]: The `exec_pledges` argument can be omitted with the `System::pledge` wrapper.

```suggestion
    TRY(Core::System::pledge("stdio cpath rpath"));
```

**Comment by bgianfo** [comment]
> There's certainly an opportunity to create a `FilePermissionParser` class to be shared by these utilities but I'm not sure where it should belong.

Sounds like `LibCore` would be a good home for such a class.

File: Userland/Utilities/mkdir.cpp [hunk 2]
```diff
@@ -8,27 +8,35 @@
...
-    args_parser.parse(argc, argv);
+    args_parser.parse(arguments);
 
-    // FIXME: Support -m/--mode option
-    mode_t mode = 0755;
+    mode_t default_mode = 0755;
+    mode_t mode = default_mode;
+
+    if (!mode_string.is_empty()) {
+        if (sscanf(mode_string.characters(), "%ho", &mode) != 1) {
```
Comment (bgianfo) [inline Userland/Utilities/mkdir.cpp:35]: We have `AK::StringUtils::convert_to_uint_from_hex(..)` and friends, maybe we should have `AK::StringUtils::convert_to_uint_from_octal(..)`?

Comment (xavier) [reply Userland/Utilities/mkdir.cpp:35]: Agreed, I had a peek into that module to circumvent the use of `sscanf` but couldn't find it. I'll look into it.

**Comment by xavier** [comment]
> Sounds like `LibCore` would be a good home for such a class.

I might take a shot a this too but it will take some time as I'm still brushing up my C++ skills.

This change would involve quite a bit of refactoring of the chmod utility. Should this be done in a separate PR?  
If you're ok with expanding the scope of this PR, that's fine with me.

**Comment by bgianfo** [comment]
Let's put it in a separate PR, no reason to block this PR on that refactoring.

**Pull request merged**